packages/enzyme-test-suite/test/RSTTraversal-spec.jsx

```diff
@@ -97,71 +68,69 @@ describe('RSTTraversal', () => {
-      expect(nodeHasProperty(<div foo={Infinity} />, 'foo', 'Infinity')).to.equal(true);
-      expect(nodeHasProperty(<div foo={-Infinity} />, 'foo', '-Infinity')).to.equal(true);
+    it('should parse numeric literals', () => {
+      expect(nodeHasProperty(<div foo={2.3} />, 'foo', 2.3)).to.equal(true);
+      expect(nodeHasProperty(<div foo={2} />, 'foo', 2)).to.equal(true);
+      expect(nodeHasProperty(<div foo={2} />, 'foo', '2abc')).to.equal(false);
+      expect(nodeHasProperty(<div foo={2} />, 'foo', 'abc2')).to.equal(false);
+      expect(nodeHasProperty(<div foo={-2} />, 'foo', -2)).to.equal(true);
+      expect(nodeHasProperty(<div foo={2e8} />, 'foo', 2e8)).to.equal(true);
+      expect(nodeHasProperty(<div foo={Infinity} />, 'foo', Infinity)).to.equal(true);
+      expect(nodeHasProperty(<div foo={-Infinity} />, 'foo', -Infinity)).to.equal(true);
     });
 
     it('should parse zeroes properly', () => {
-      expect(nodeHasProperty(<div foo={0} />, 'foo', '0')).to.equal(true);
-      expect(nodeHasProperty(<div foo={-0} />, 'foo', '-0')).to.equal(true);
-      expect(nodeHasProperty(<div foo={1} />, 'foo', '0')).to.equal(false);
-      expect(nodeHasProperty(<div foo={2} />, 'foo', '-0')).to.equal(false);
+      expect(nodeHasProperty(<div foo={0} />, 'foo', 0)).to.equal(true);
+      expect(nodeHasProperty(<div foo={0} />, 'foo', +0)).to.equal(true);
+      expect(nodeHasProperty(<div foo={-0} />, 'foo', -0)).to.equal(true);
+      expect(nodeHasProperty(<div foo={-0} />, 'foo', 0)).to.equal(false);
+      expect(nodeHasProperty(<div foo={0} />, 'foo', -0)).to.equal(false);
+      expect(nodeHasProperty(<div foo={1} />, 'foo', 0)).to.equal(false);
+      expect(nodeHasProperty(<div foo={2} />, 'foo', -0)).to.equal(false);
     });
 
     it('should work with empty strings', () => {
+      expect(nodeHasProperty(<div foo="" />, 'foo', '')).to.equal(true);
       expect(nodeHasProperty(<div foo={''} />, 'foo', '')).to.equal(true);
-      expect(nodeHasProperty(<div foo={''} />, 'foo', '""')).to.equal(true);
       expect(nodeHasProperty(<div foo={'bar'} />, 'foo', '')).to.equal(false);
-      expect(nodeHasProperty(<div foo={'bar'} />, 'foo', '""')).to.equal(false);
     });
 
     it('should work with NaN', () => {
-      expect(nodeHasProperty(<div foo={NaN} />, 'foo', 'NaN')).to.equal(true);
-      expect(nodeHasProperty(<div foo={0} />, 'foo', 'NaN')).to.equal(false);
+      expect(nodeHasProperty(<div foo={NaN} />, 'foo', NaN)).to.equal(true);
+      expect(nodeHasProperty(<div foo={0} />, 'foo', NaN)).to.equal(false);
     });
 
     it('should work with null', () => {
-      expect(nodeHasProperty(<div foo={null} />, 'foo', 'null')).to.equal(true);
-      expect(nodeHasProperty(<div foo={0} />, 'foo', 'null')).to.equal(false);
+      expect(nodeHasProperty(<div foo={null} />, 'foo', null)).to.equal(true);
+      expect(nodeHasProperty(<div foo={0} />, 'foo', null)).to.equal(false);
```

Comment:
Let's also add a pair of assertions that null won't equal undefined?

Reply:
What's the expected behavior with `undefined`? It looks like it returns [whether `nodeProps` has an own property matching the name](https://github.com/airbnb/enzyme/blob/da9140713a946ef3bef23952e693bc917166a0b3/packages/enzyme/src/Utils.js#L355), which means `undefined` ends up matching since `foo` exists.

Is that expected?

Reply:
`nodeHasProperty(<div foo={undefined} />, 'foo', undefined)` should be `true`, `nodeHasProperty(<div foo={undefined} />, 'foo', null)` should be `false`, `nodeHasProperty(<div foo={null} />, 'foo', undefined)` should be `false`.

Reply:
Right now `nodeHasProperty(<div foo={null} />, 'foo', undefined)` returns `true` because it's defaulting to the own-property check. I think this behavior existed before this PR too, is that a bug? I'm not sure if that will break some existing tests.

Reply:
ping @ljharb ^ I think that's the last point of clarification I need before this is good to go.

Reply:
In the interest of time, i'm going to move forward w/ this PR. I've verified that this is the current behavior. If we would like to clarify it / change it in an upcoming PR, i'm fine with that. @aweary thanks for being so thorough here.